src/charm_state.py

```diff
@@ -4,18 +4,67 @@
 """This module defines the CharmState class which represents the state of the Flask charm."""
 
 import datetime
+import itertools
 import pathlib
 
 from ops.charm import CharmBase
+from pydantic import (  # pylint: disable=no-name-in-module
+    BaseModel,
+    Extra,
+    Field,
+    ValidationError,
+    validator,
+)
 
 from charm_types import WebserverConfig
+from exceptions import CharmConfigInvalidError
+
+
+class FlaskConfig(BaseModel, extra=Extra.allow):  # pylint: disable=too-few-public-methods
+    """Represent Flask builtin configuration values.
+
+    Attrs:
+        env: what environment the Flask app is running in, by default it's 'production'.
+        debug: whether Flask debug mode is enabled.
+        secret_key: a secret key that will be used for securely signing the session cookie
+            and can be used for any other security related needs by your Flask application.
+        permanent_session_lifetime: set the cookie’s expiration to this number of seconds in the
+            Flask application permanent sessions.
+        application_root: inform the Flask application what path it is mounted under by the
+            application / web server.
+        session_cookie_secure: set the secure attribute in the Flask application cookies.
+        preferred_url_scheme: use this scheme for generating external URLs when not in a request
+            context in the Flask application.
+    """
+
+    env: str = Field(None, min_length=1)
+    debug: bool = Field(None)
+    secret_key: str = Field(None, min_length=1)
```

Comment:
I think that if no secret_key is specified we have to generate one as Flask will block Session if that's not set

Reply:
I believe that charm should be consistent with Flask. If the charm does not have a secret key set, the charm managed flask should also not have a secret key set.

Reply:
Proposal is to implement this in a follow up story, could you create that story and we can make it out of scope for this pr?

Reply:
Ok fine, I made a story